generative_ai/embedding.py

```diff
@@ -13,18 +13,28 @@
 # limitations under the License.
 
 # [START aiplatform_sdk_embedding]
-from vertexai.language_models import TextEmbeddingModel
+from typing import List
 
-
-def text_embedding(text: str = "What is life?") -> list:
-    """Text embedding with a Large Language Model."""
-    model = TextEmbeddingModel.from_pretrained("textembedding-gecko@001")
-    embeddings = model.get_embeddings([text])
-    vector = embeddings[0].values
-    print(f"Length of Embedding Vector: {len(vector)}")
-    return vector
+from vertexai.language_models import TextEmbeddingInput, TextEmbeddingModel
 
 
+def embed_text(
+    texts: List[str] = ["banana muffins? ", "banana bread? banana muffins?"],
+    task_types: List[str] = ["QUESTION_ANSWERING", "FACT_VERIFICATION"],
+    model_name: str = "text-embedding-preview-0409",
+) -> List[List[float]]:
+    """Embeds texts with a pre-trained, foundational model."""
+    model = TextEmbeddingModel.from_pretrained(model_name)
+    elastic = model_name in {
+        "text-embedding-preview-0409",
+        "text-multilingual-embedding-preview-0409",
+    }
+    inputs = [TextEmbeddingInput(*z) for z in zip(texts, task_types)]
+    kwargs = dict(output_dimensionality=192) if elastic else {}
```

Comment:
Does this field need to be included in kwargs? I would recommend explicitly setting it in the `get_embeddings()` method. You can also pass it as an argument to the `embed_text()` method.

Reply:
You cannot set output_dimensionality for unsupported models and unsupported vertex ai platform versions. You would pass {} (empty kwargs) for some unsupported cases. I found that my Python samples are running ~6 week-old vertex ai platform sdk, which may end up with unknown args errors.